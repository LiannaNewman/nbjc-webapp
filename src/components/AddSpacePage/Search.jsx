import React, { useState } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { withStyles } from '@material-ui/core/styles';
import useMediaQuery from '@material-ui/core/useMediaQuery';

import Button from '@material-ui/core/Button';
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';

import states from '../../api/states';
import { addSpaceProps as addSpacePropTypes } from '../../types';

const styles = (theme) => ({
  pageTitle: {
    marginBottom: 10,
  },
  pageSubtitle: {
    [theme.breakpoints.up('mobile')]: {
      marginBottom: 60,
    },
    [theme.breakpoints.up('xs')]: {
      marginBottom: 40,
    },
  },
  form: {
    [theme.breakpoints.up('mobile')]: {
      margin: 0,
    },
    [theme.breakpoints.up('xs')]: {
      margin: '0 20px',
    },
  },
  inputLabel: {
    marginBottom: '10px',
    color: '#000',
  },
  inputField: {
    display: 'block',
    '& input': {
      [theme.breakpoints.up('mobile')]: {
        minWidth: '326px',
      },
    },
  },
  section: {
    [theme.breakpoints.up('mobile')]: {
      marginTop: 30,
    },
    [theme.breakpoints.up('xs')]: {
      marginTop: 15,
    },
  },
  stateLabel: {
    padding: '0px 12px',
  },
  smallFormControl: {
    width: 150,
  },
  formControl: {
    marginTop: '15px',
    [theme.breakpoints.up('xs')]: {
      display: 'block',
    },
    [theme.breakpoints.up('mobile')]: {
      display: 'inline-block',
      marginRight: 22,
    },
  },
  footer: {
    display: 'block',
  },
  submitButton: {
    float: 'right',
    marginTop: 20,
  },
});

const Search = ({
  classes,
  onNext,
  disableNext,
  addSpaceProps,
}) => {
  const matches = useMediaQuery('(min-width:376px)');
  const [formValues, setFormValues] = useState({
    name: addSpaceProps.name || '',
    city: addSpaceProps.city || '',
    zipcode: addSpaceProps.zipcode || '',
    state: addSpaceProps.state || '',
  });
  const onChange = (e) => {
    setFormValues({
      ...formValues,
      [e.target.name]: e.target.value,
    });
  };

  const onSearch = (e) => {
    e.preventDefault();
    onNext(formValues);
  };

  const validateForm = () => formValues.name.length > 0
    && formValues.city.length > 0
    && formValues.state.length === 2;

  return (
    <>
      <Typography
        variant={matches ? 'h2' : 'h4'}
        align="center"
        className={classes.pageTitle}
      >
        Add a Space
      </Typography>
      <Typography
        variant={matches ? 'h4' : 'subtitle1'}
        align="center"
        className={classes.pageSubtitle}
      >
        Add your favorite space to The Lavender Book. The information you enter will be
        reviewed by an administrator. Thank you for helping your community find
        and enjoy spaces.
      </Typography>
      <form onSubmit={onSearch} className={classes.form}>
        <InputLabel type="inputLabel" className={classes.inputLabel}>
          <Typography variant="h6">What&apos;s the space?</Typography>
        </InputLabel>
        <TextField
          type="input"
          label="Name of Space"
          variant="outlined"
          helperText="Required"
          className={classes.inputField}
          name="name"
          onChange={onChange}
          value={formValues.name}
          fullWidth={!matches}
          data-testid="addspace-name"
          required
        />
        <InputLabel
          type="inputLabel"
          className={cx(classes.inputLabel, classes.section)}
        >
          <Typography variant="h6">Space Location</Typography>
        </InputLabel>
        <TextField
          type="input"
          label="City"
          variant="outlined"
          helperText="Required"
          className={classes.inputField}
          name="city"
          onChange={onChange}
          value={formValues.city}
          fullWidth={!matches}
          data-testid="addspace-city"
          required
        />
        <FormControl className={classes.formControl}>
          <InputLabel type="inputLabel" className={classes.stateLabel} required>
            State
          </InputLabel>
          <Select
            type="input"
            label="City"
            variant="outlined"
            className={classes.smallFormControl}
            name="state"
            onChange={onChange}
            value={formValues.state}
            data-testid="addspace-state"
            required
          >
            {states.map((state) => (
              <MenuItem value={state.abbreviation} key={state.abbreviation}>
                {state.abbreviation}
              </MenuItem>
            ))}
          </Select>
        </FormControl>
        <FormControl className={classes.formControl}>
          <TextField
            type="input"
            label="Zipcode"
            variant="outlined"
            helperText="Optional"
            className={classes.smallFormControl}
            onChange={onChange}
            value={formValues.zipcode}
            name="zipcode"
            data-testid="addspace-zipcode"
          />
        </FormControl>
        <div className={classes.footer}>
          <Button
            type="submit"
            variant="contained"
            color="primary"
            onClick={onSearch}
            className={cx(classes.inputField, classes.submitButton)}
            fullWidth={!matches}
            onChange={onChange}
            disabled={!validateForm() || disableNext}
            data-testid="addspace-search-next"
            disableElevation
          >
            Next
          </Button>
        </div>
      </form>
    </>
  );
};

Search.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  onNext: PropTypes.func,
  disableNext: PropTypes.bool,
  addSpaceProps: PropTypes.shape(addSpacePropTypes),
};

Search.defaultProps = {
  onNext: () => {},
  disableNext: false,
  addSpaceProps: {},
};

export default withStyles(styles)(Search);
