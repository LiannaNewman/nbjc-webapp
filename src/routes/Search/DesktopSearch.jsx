import React from 'react';

const DesktopSearch = () => (
  <>
  </>
);

export default DesktopSearch;
