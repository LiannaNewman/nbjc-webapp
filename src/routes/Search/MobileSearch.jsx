import React, { useState } from 'react';
import { OutlinedInput, InputAdornment, IconButton } from '@material-ui/core';
import SearchIcon from '@material-ui/icons/Search';

import ClearOutlinedIcon from '@material-ui/icons/ClearOutlined';
import FilterListOutlinedIcon from '@material-ui/icons/FilterListOutlined';

import FilterDialog from '../../components/FilterDialog';

const MobileSearch = () => {
  const defaultFilters = {
    stars: 3,
    price: 2,
    distance: 5,
  };

  const [searchText, setSearchText] = useState('');
  const [openFilter, setOpenFilter] = useState(false);
  const [filters, setFilters] = useState(defaultFilters);
  const handleChange = (event) => {
    setSearchText(event.target.value);
  };

  return (
    <>
      <form>
        <OutlinedInput
          type="text"
          endAdornment={(
            <InputAdornment position="end">
              <IconButton>
                {searchText.length > 0 ? (
                  <ClearOutlinedIcon onClick={() => setSearchText('')} />
                ) : (
                  <SearchIcon />
                )}
              </IconButton>
            </InputAdornment>
          )}
          variant="outlined"
          fullWidth
          style={{ margin: 8 }}
          value={searchText}
          onChange={handleChange}
        />
        <FilterListOutlinedIcon onClick={() => setOpenFilter(true)} />
      </form>
      <FilterDialog
        open={openFilter}
        onClose={() => setOpenFilter(false)}
        defaultFilters={filters}
        setFilters={(changedFilters) => setFilters(changedFilters)}
      />
    </>
  );
};

MobileSearch.propTypes = {};

export default MobileSearch;
