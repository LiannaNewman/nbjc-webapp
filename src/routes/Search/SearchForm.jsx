import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import useMediaQuery from '@material-ui/core/useMediaQuery';

import {
  Button,
  TextField,
  Typography,
  InputLabel,
} from '@material-ui/core';

import Autocomplete from '@material-ui/lab/Autocomplete';
import { withStyles } from '@material-ui/core/styles';

import FilterListOutlinedIcon from '@material-ui/icons/FilterListOutlined';

import { searchProps, chipType } from '../../types';

import ChipFilters from '../../components/ChipFilters';
import FilterDialog from '../../components/FilterDialog';
import ErrorSnackbar from '../../components/ErrorSnackbar';
import { getCategories, getSpacesByName } from '../../api';

const styles = (theme) => ({
  form: {
    margin: '10px 0px',
  },
  filterWrapper: {
    display: 'flex',
    [theme.breakpoints.up('mobile')]: {
      flexWrap: 'wrap',
    },
  },
  filterButton: {
    margin: '10px 0px',
  },
  filterDialog: {
    [theme.breakpoints.up('mobile')]: {
      width: '100%',
      marginBottom: 40,
      marginTop: 10,
    },
  },
});

const SearchForm = ({
  classes,
  onSearch,
  onFilterApplied,
  searchCriteria,
  chips,
}) => {
  const matches = useMediaQuery('(min-width:376px)');
  const [formValues, setFormValues] = useState({
    name: '',
    location: '',
    category: {},
    indicators: chips.map((chip) => ({
      ...chip,
      isSelected: false,
    })),
  });

  const [openFilter, setOpenFilter] = useState(false);
  const [showError, setShowError] = useState(false);

  /** Start Name Autocomplete */
  const [nameText, setNameText] = useState('');
  const [spaceNames, setSpaceNames] = useState([]);
  const onNameTextChange = (e) => {
    setNameText(e.target.value);
  };
  useEffect(() => {
    const fetchData = async () => {
      const result = await getSpacesByName({ name: nameText });
      setSpaceNames(result);
    };
    fetchData();
  }, [nameText]);
  const onNameChange = (e, value) => {
    if (value === null) {
      setNameText('');
    }
    setFormValues({
      ...formValues,
      name: value,
    });
  };
  /** End Name Autocomplete */

  /** Start Category Autocomplete */
  const [categoryText, setCategoryText] = useState('');
  const [categories, setCategories] = useState([]);

  const onCategoryTextChange = (e) => {
    setCategoryText(e.target.value);
  };

  useEffect(() => {
    const fetchData = async () => {
      const result = await getCategories({ searchTerm: categoryText });
      setCategories(result);
    };
    fetchData();
  }, [categoryText]);

  const onCategoryChange = (e, value) => {
    if (value === null) {
      setCategoryText('');
    }
    setFormValues({
      ...formValues,
      category: value,
    });
  };
  /** End Category Autocomplete */
  const handleChange = (e) => {
    setFormValues({
      ...formValues,
      [e.target.name]: e.target.value,
    });
  };

  const handleIndicatorSelect = (value) => {
    setFormValues({
      ...formValues,
      indicators: formValues.indicators.map((chip) => ({
        ...chip,
        ...(chip.value === value
          ? { isSelected: !chip.isSelected }
          : {}),
      })),
    });
  };

  const onSearchSubmit = (e) => {
    e.preventDefault();
    if (
      formValues.name.length === 0
      && formValues.location.length === 0
      && formValues.category.length === 0
    ) {
      setShowError(true);
    } else {
      onSearch(formValues);
    }
  };

  const filters = {
    stars: searchCriteria.rating || 0,
    distance: searchCriteria.distance || 0,
    price: searchCriteria.price || 0,
  };

  const setFilters = (changedFilters) => {
    onFilterApplied('distance', changedFilters.distance);
    onFilterApplied('price', changedFilters.price);
    onFilterApplied('rating', changedFilters.stars);
  };
  return (
    <>
      <form className={classes.form} onSubmit={onSearchSubmit}>
        <Typography variant={matches ? 'h2' : 'h4'} align="center">
          Search for a Space
        </Typography>
        <Typography
          variant={matches ? 'h4' : 'subtitle1'}
          align="center"
        >
          Search for spaces by name, location, category, or attributes.
        </Typography>
        <InputLabel type="inputLabel" className={classes.inputLabel}>
          <Typography variant="h6">What&apos;s the space?</Typography>
        </InputLabel>
        <Autocomplete
          getOptionSelected={(option, value) => option.name === value.name}
          getOptionLabel={(option) => option.name}
          options={spaceNames}
          style={{ width: 300 }}
          onChange={onNameChange}
          renderInput={(params) => (
            <TextField
              type="text"
              variant="outlined"
              style={{ margin: '8px 0px' }}
              value={nameText}
              onChange={onNameTextChange}
              placeholder="Space Name"
              helperText="Optional"
              name="name"
              autoFocus
              // eslint-disable-next-line react/jsx-props-no-spreading
              {...params}
            />
          )}
        />
        <InputLabel type="inputLabel" className={classes.inputLabel}>
          <Typography variant="h6">Where are you looking?</Typography>
        </InputLabel>
        <TextField
          type="text"
          variant="outlined"
          fullWidth={!matches}
          style={{ margin: '8px 0px' }}
          value={formValues.location}
          onChange={handleChange}
          placeholder="City, State"
          helperText="Optional"
          name="location"
          autoFocus
        />
        <InputLabel type="inputLabel" className={classes.inputLabel}>
          <Typography variant="h6">What type of Space are you looking for? </Typography>
        </InputLabel>
        <Autocomplete
          getOptionSelected={(option, value) => option.title === value.title}
          getOptionLabel={(option) => option.title}
          options={categories}
          style={{ width: 300 }}
          onChange={onCategoryChange}
          renderInput={(params) => (
            <TextField
              type="text"
              variant="outlined"
              style={{ margin: '8px 0px' }}
              value={categoryText}
              onChange={onCategoryTextChange}
              placeholder="Space Category"
              helperText="Optional"
              name="category"
              autoFocus
              // eslint-disable-next-line react/jsx-props-no-spreading
              {...params}
            />
          )}
        />

        <div className={classes.filterWrapper}>
          { false
            && (
            <FilterDialog
              open={openFilter}
              onClose={() => setOpenFilter(false)}
              onToggle={() => setOpenFilter(!openFilter)}
              defaultFilters={filters}
              setFilters={(changedFilters) => setFilters(changedFilters)}
              type={matches ? 'desktop' : 'mobile'}
              overrideClasses={{ root: classes.filterDialog }}
            />
            )}
          {!matches && false && (
            <FilterListOutlinedIcon
              onClick={() => setOpenFilter(true)}
              className={classes.filterButton}
            />
          )}
          <InputLabel type="inputLabel" className={classes.inputLabel}>
            <Typography variant="h6">Select all that apply</Typography>
          </InputLabel>
          <ChipFilters
            chips={formValues.indicators}
            onChipSelected={(i) => handleIndicatorSelect(chips[i].value)}
          />
        </div>
        <Button
          type="submit"
          variant="contained"
          color="secondary"
          fullWidth={!matches}
          data-testid="search-searchform-submit"
          disableElevation
        >
          Search
        </Button>
      </form>
      <ErrorSnackbar
        snackbarOpen={showError}
        onClose={() => setShowError(false)}
        body="You must enter at least one piece of infomration in order to search for a space."
        showSupport={false}
      />
    </>
  );
};

SearchForm.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  onSearch: PropTypes.func.isRequired,
  onFilterApplied: PropTypes.func.isRequired,
  searchCriteria: PropTypes.shape(searchProps).isRequired,
  chips: PropTypes.arrayOf(PropTypes.shape(chipType)).isRequired,
};

SearchForm.defaultProps = {};

export default withStyles(styles)(SearchForm);
