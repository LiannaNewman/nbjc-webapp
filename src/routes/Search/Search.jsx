import React, { useState, useEffect } from 'react';

import PropTypes from 'prop-types';
import { geolocated, geoPropTypes } from 'react-geolocated';
import cx from 'classnames';

import useMediaQuery from '@material-ui/core/useMediaQuery';
import { withStyles } from '@material-ui/core/styles';

import Button from '@material-ui/core/Button';
import CircularProgress from '@material-ui/core/CircularProgress';
import Typography from '@material-ui/core/Typography';
import SearchIcon from '@material-ui/icons/Search';

import BusinessCard from '../../components/BusinessCard';
import FilterDialog from '../../components/FilterDialog';
import { getAllIndicators } from '../../api';

import useSearch from './hooks/useSearch';
import SearchForm from './SearchForm';

const styles = (theme) => ({
  result: {
    margin: '10px 0px 40px 0px',
  },
  content: {
    margin: '0 15px',
    [theme.breakpoints.up('mobile')]: {
      margin: '0 100px',
    },
  },
  searchResultsWrapper: {
    display: 'grid',
    gridTemplateColumns: 'repeat(auto-fill, 350px)',
    gridGap: 50,
  },
  searchResult: {
    [theme.breakpoints.up('xs')]: {
      maxWidth: '350px',
    },
    [theme.breakpoints.up('mobile')]: {
      width: '100%',
    },
  },
  emptyStateWrapper: {
    marginTop: 60,
  },
  emptyStateIcon: {
    width: 55,
    height: 55,
    margin: '0 auto',
  },
  emptyStateFooter: {
    marginTop: 80,
    display: 'flex',
    justifyContent: 'center',
    '& a:first-child': {
      marginRight: 20,
    },
  },
  filterButtonWrapper: {
    display: 'flex',
  },
  searchresultsText: {
    flexGrow: 2,
  },
  filterButton: {
    width: '100px !important',
    height: 36,
  },
});

const Search = ({ classes, coords }) => {
  const matches = useMediaQuery('(min-width:376px)');
  const [indicators, setIndicators] = useState([]);
  const [openFilter, setOpenFilter] = useState(false);

  useEffect(() => {
    async function fetchData() {
      const data = await getAllIndicators();
      setIndicators(data);
    }
    try {
      fetchData();
    } catch (err) {
      // todo: retry?
      console.log(err);
    }
  }, []);

  const {
    updateSearch,
    updateFilters,
    search,
    searchResults,
    loading,
    pagination,
    userLocation,
  } = useSearch({ userCoords: coords, indicators });
  const onSearchSubmit = async (searchTerm) => {
    updateSearch(searchTerm);
  };

  const onFilterApplied = (filter, value) => {
    updateSearch(filter, value);
  };

  const filters = {
    stars: search.rating || 0,
    distance: search.distance || 0,
    price: search.price || 0,
  };

  const setFilters = (changedFilters) => {
    updateFilters('distance', changedFilters.distance);
    updateFilters('price', changedFilters.price);
    updateFilters('rating', changedFilters.stars);
  };

  const getResultsString = () => {
    let text = `${pagination.total_count} results found for`;
    let comma = '';
    if (search.searchTerm) {
      text = `${text} ${search.searchTerm}`;
      comma = ',';
    }
    if (search.category) {
      text = `${text}${comma} ${search.category}`;
      comma = ',';
    }

    const indicatorNames = indicators
      .filter((indicator) => search.indicators.includes(indicator.value))
      .map((indicator) => indicator.name);

    if (indicatorNames.length < 3) {
      text = `${text}${comma} ${indicatorNames.join(' and ')}`;
    } else {
      text = `${text}${comma} ${indicatorNames[0]} and ${indicatorNames.length - 1} more`;
    }
    return text;
  };

  return (
    <div className={classes.content}>
      {indicators.length > 0 && searchResults === null && (
        <SearchForm
          chips={indicators}
          onSearch={onSearchSubmit}
          onFilterApplied={onFilterApplied}
          searchCriteria={search}
          location={userLocation.address}
        />
      )}
      <div
        className={cx(classes.resultsWrapper, {
          [classes.desktop]: matches,
        })}
      >
        {searchResults !== null && searchResults.length > 0 && (
          <>
            <div className={classes.filterButtonWrapper}>
              <Typography variant="h6" className={classes.searchresultsText}>
                {getResultsString()}
              </Typography>
              <Button variant="outlined" onClick={() => setOpenFilter(!openFilter)} color="primary" className={classes.filterButton}>FILTER</Button>
            </div>
            <div className={classes.filterWrapper}>
              {openFilter && (
                <FilterDialog
                  open={openFilter}
                  onClose={() => setOpenFilter(false)}
                  onToggle={() => setOpenFilter(!openFilter)}
                  defaultFilters={filters}
                  setFilters={(changedFilters) => setFilters(changedFilters)}
                  type={matches ? 'desktop' : 'mobile'}
                  overrideClasses={{ root: classes.filterDialog }}
                />
              )}
            </div>
          </>
        )}
        <div className={classes.searchResultsWrapper}>
          {searchResults !== null && searchResults.length > 0
            && searchResults.map((result) => (
              <div className={classes.searchResult} key={result.id}>
                <BusinessCard
                  business={result}
                  key={result.id}
                  overrideClasses={{ root: classes.result }}
                />
              </div>
            ))}
        </div>
        {searchResults !== null
          && search.searchTerm !== null
          && searchResults.length === 0
          && !loading
          && (
            <div className={classes.emptyStateWrapper}>
              <Typography variant={matches ? 'h2' : 'h4'} align="center">
                No Results
              </Typography>
              <div className={classes.emptyStateIcon}>
                <SearchIcon color="primary" fontSize="large" className={classes.emptyStateIcon} />
              </div>
              <Typography variant={matches ? 'h4' : 'subtitle1'} align="center">
                We couldn’t find what you’re looking for.
                Please try again or add a space to OurGuide.
              </Typography>
              <div className={classes.emptyStateFooter}>
                <Button variant="outlined" href="/" color="secondary">
                  Home
                </Button>
                <Button variant="contained" href="/space/new" color="secondary" disableElevation>
                  Add space
                </Button>
              </div>
            </div>
          )}
        {(loading || indicators.length < 1) && <CircularProgress color="secondary" />}
      </div>
    </div>
  );
};

Search.props = {
  classes: PropTypes.shape({}),
};

Search.props = { ...Search.props, ...geoPropTypes };

export default geolocated()(withStyles(styles)(Search));
