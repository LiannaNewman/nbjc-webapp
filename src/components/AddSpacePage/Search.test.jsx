import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

import Search from './Search';

describe('Search', () => {
  test('renders with default values', () => {
    const { asFragment } = render(
      <Search />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders the form with validation and passes data on submit', () => {
    const onNext = jest.fn();
    render(
      <Search onNext={onNext} />,
    );
    const nextButton = screen.getByText('Next').closest('button');
    expect(nextButton).toBeDisabled();
    // enter form details
    const name = screen.getByTestId('addspace-name').querySelector('input');
    fireEvent.change(name, { target: { value: 'Test' } });
    const city = screen.getByTestId('addspace-city').querySelector('input');
    fireEvent.change(city, { target: { value: 'test city' } });
    const state = screen.getByTestId('addspace-state').querySelector('input');
    fireEvent.change(state, { target: { value: 'NY' } });
    expect(nextButton).not.toBeDisabled();
    const zip = screen.getByTestId('addspace-zipcode').querySelector('input');
    fireEvent.change(zip, { target: { value: '10001' } });

    fireEvent.click(nextButton);
    expect(onNext.mock.calls.length).toBe(1);
    expect(onNext.mock.calls[0][0]).toEqual({
      name: 'Test',
      city: 'test city',
      state: 'NY',
      zipcode: '10001',
    });
  });
});
