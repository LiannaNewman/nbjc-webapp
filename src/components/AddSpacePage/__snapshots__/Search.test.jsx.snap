// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders with default values 1`] = `
<DocumentFragment>
  <h4
    class="MuiTypography-root Search-pageTitle-1 MuiTypography-h4 MuiTypography-alignCenter"
  >
    Add a Space
  </h4>
  <h6
    class="MuiTypography-root Search-pageSubtitle-2 MuiTypography-subtitle1 MuiTypography-alignCenter"
  >
    Add your favorite space to OurGuide. The information you enter will be reviewed by an administrator. Thank you for helping your community find and enjoy spaces.
  </h6>
  <form
    class="Search-form-3"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root Search-inputLabel-4 MuiInputLabel-animated"
      type="inputLabel"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        What's the space?
      </h6>
    </label>
    <div
      class="MuiFormControl-root MuiTextField-root Search-inputField-5 MuiFormControl-fullWidth"
      data-testid="addspace-name"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink="false"
      >
        Name of Space
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          name="name"
          required=""
          type="input"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-14"
          >
            <span>
              Name of Space *
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained Mui-required"
      >
        Required
      </p>
    </div>
    <label
      class="MuiFormLabel-root MuiInputLabel-root Search-inputLabel-4 Search-section-6 MuiInputLabel-animated"
      type="inputLabel"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        Space Location
      </h6>
    </label>
    <div
      class="MuiFormControl-root MuiTextField-root Search-inputField-5 MuiFormControl-fullWidth"
      data-testid="addspace-city"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink="false"
      >
        City
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          name="city"
          required=""
          type="input"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-14"
          >
            <span>
              City *
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained Mui-required"
      >
        Required
      </p>
    </div>
    <div
      class="MuiFormControl-root Search-formControl-9"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root Search-stateLabel-7 MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink="false"
        type="inputLabel"
      >
        State
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root Search-smallFormControl-8 MuiInputBase-formControl"
        data-testid="addspace-state"
        required=""
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="mui-component-select-state"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
          id="mui-component-select-state"
          role="button"
          tabindex="0"
        >
          <span>
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          name="state"
          required=""
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-14"
          >
            <span>
              City
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root Search-formControl-9"
    >
      <div
        class="MuiFormControl-root MuiTextField-root Search-smallFormControl-8"
        data-testid="addspace-zipcode"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Zipcode
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="zipcode"
            type="input"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-14"
            >
              <span>
                Zipcode
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
        >
          Optional
        </p>
      </div>
    </div>
    <div
      class="Search-footer-10"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained Search-inputField-5 Search-submitButton-11 MuiButton-containedSecondary MuiButton-disableElevation Mui-disabled MuiButton-fullWidth Mui-disabled"
        data-testid="addspace-search-next"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Next
        </span>
      </button>
    </div>
  </form>
</DocumentFragment>
`;
